src/plugins/issue-pr-link/index.js

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Extract issue numbers from PR title
+ * @param {string} title PR title
+ * @returns {number[]} Array of issue numbers
+ * @private
+ */
+function extractIssueNumbers(title) {
+    const matches = [];
+    let match;
+
+    // Reset regex lastIndex to ensure we start from the beginning
+    ISSUE_REFERENCE_REGEX.lastIndex = 0;
+
+    while ((match = ISSUE_REFERENCE_REGEX.exec(title)) !== null && matches.length < MAX_ISSUES_PER_PR) {
+        const issueNumber = parseInt(match[1], 10);
+        if (!matches.includes(issueNumber)) {
+            matches.push(issueNumber);
+        }
+    }
+
+    return matches;
+}
+
+/**
+ * Create the comment message for the issue
+ * @param {string} prUrl URL of the pull request
+ * @param {string} prTitle Title of the pull request
+ * @param {string} prAuthor Author of the pull request
+ * @returns {string} comment message
+ * @private
+ */
+function createCommentMessage(prUrl, prTitle, prAuthor) {
+    return `👋 Hi! This issue is being addressed in pull request ${prUrl}. Thanks, @${prAuthor}!
+
+[//]: # (issue-pr-link)`;
+}
+
+/**
+ * Check if an issue exists and is open
+ * @param {ProbotContext} context Probot context object
+ * @param {number} issueNumber Issue number to check
+ * @returns {Promise<boolean>} True if issue exists and is open
+ * @private
+ */
+async function isIssueOpenAndExists(context, issueNumber) {
+    try {
+        const { data: issue } = await context.octokit.issues.get(
+            context.repo({ issue_number: issueNumber })
+        );
+        return issue.state === "open";
+    } catch {
+        // Issue doesn't exist or we don't have access
+        return false;
+    }
+}
+
+/**
+ * Check if we already commented on this issue for this PR
+ * @param {ProbotContext} context Probot context object
+ * @param {number} issueNumber Issue number
+ * @param {number} prNumber PR number
+ * @returns {Promise<boolean>} True if we already commented
+ * @private
+ */
+async function hasExistingComment(context, issueNumber, prNumber) {
+    try {
+        const { data: comments } = await context.octokit.issues.listComments(
+            context.repo({ issue_number: issueNumber })
+        );
+
+        const botComments = comments.filter(comment => 
+            comment.user.type === "Bot" && 
+            comment.body.includes("[//]: # (issue-pr-link)") &&
+            comment.body.includes(`/pull/${prNumber}`)
+        );
+
+        return botComments.length > 0;
+    } catch {
+        // If we can't check comments, assume we haven't commented
+        return false;
+    }
+}
+
+/**
+ * Comment on issues referenced in the PR title
+ * @param {ProbotContext} context Probot context object
+ * @returns {Promise<void>}
+ * @private
+ */
+async function commentOnReferencedIssues(context) {
+    const { payload } = context;
+    const pr = payload.pull_request;
+    
+    if (!pr || !pr.title) {
+        return;
+    }
+
+    const issueNumbers = extractIssueNumbers(pr.title);
```

Comment:
```suggestion
    const issueNumbers = extractIssueNumbers(pr.body);
```

Looking at the Octokit and GitHub REST APIs, I believe it should be `pr.body`, not `pr.title`.

- https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request
- https://octokit.github.io/rest.js/v21/#pulls-get

<img width="636" height="234" alt="image" src="https://github.com/user-attachments/assets/c97e8f75-638c-4e40-b8b1-a5fbd8c74826" />